templates/UmbracoPackageRcl/UmbracoPackage.csproj

```diff
@@ -24,4 +24,21 @@
     <PackageReference Include="Umbraco.Cms.Web.Website" Version="UMBRACO_VERSION_FROM_TEMPLATE" />
     <PackageReference Include="Umbraco.Cms.Web.BackOffice" Version="UMBRACO_VERSION_FROM_TEMPLATE" />
   </ItemGroup>
+
+  <ItemGroup>
+    <Content Update="package.json" Pack="False" />
+    <Content Update="package-lock.json" Pack="False" />
+    <Content Update="tsconfig.json" Pack="False" />
+  </ItemGroup>
+
+  <Target Name="NpmInstall" AfterTargets="Restore;Build" Inputs="package-lock.json" Outputs="node_modules\.package-lock.json">
+    <Message Text="Installing NPM packages" Importance="High" />
+    <Exec Command="npm ci --no-fund --no-audit --prefer-offline" />
```

Comment:
`npm ci` performs a clean install of all the packages, i.e. it wipes out and recreates `./node_modules/` - that shouldn't be necessary to run for every build. The ASP.NET example templates  add some conditional logic to only run when `/node_modules/` doesn't exist.

https://docs.npmjs.com/cli/v10/commands/npm-ci#description

Reply:
I'm wondering if it shouldn't be enough to run `npm install` which would make the incremental builds even faster. With modern npm, the discrepancies between the lock file and package file are less critical and you won't end up with a completely different version of a sub-sub-dependency anymore, as was the case of older npm, which spawned the need for the `ci` command in the first place. Also, if someone for some reason forgot to commit the changed lockfile, the `ci` command actually fails resulting in the entire build failing. This is not the case with `npm install`. What do you think?